docs/react-native-windows-init.md

```diff
@@ -0,0 +1,66 @@
+# React native windows initialization
+
+This document describes how the template initialization works for `react-native-windows init`.
+
+## Where is the code:
+The main logic that determines the template values and which files to apply is function [`copyProjecTemplateAndReplace` in `vnext/local-cli/generator-windows/index.js`](https://github.com/microsoft/react-native-windows/blob/master/vnext/local-cli/generator-windows/index.js#L48).
+The parameters for `init` is located in variable [`argv` in `packages/react-native-windows-init/src/Cli.ts`](https://github.com/microsoft/react-native-windows/blob/7de1f3843a9754c7fbf744bf3e556fdca78472c1/packages/react-native-windows-init/src/Cli.ts#L27)
+
+The template files can be found in [`vnext/local-cli/generator-windows/templates`](https://github.com/microsoft/react-native-windows/tree/7de1f3843a9754c7fbf744bf3e556fdca78472c1/vnext/local-cli/generator-windows/templates)
+
+## Debugging:
+To debug the template [Contributing to the CLI](https://github.com/microsoft/react-native-windows/wiki/Contributing-to-the-CLI) on our wiki.
+
+## Templating tech (Mustache)
+PR [#5126](https://github.com/microsoft/react-native-windows/pull/5126) updated the template generation for react-native-windows from regex replace to use the mustache template library. 
+### Benefits
+This has the following benefits:
+* We can use conditionals and foreach in the template allowing for better expersivity
+* This expresivity shows that we don't have to construct Xml fragments in Javascript anymore to be inserted by regex replace
+* We don't need to have have a copy and edit for large differences like we had for experimental nuget, it is now a single template for each language.
+* We can now generate a cpp template that use both nuget AND winUI3.
+
+### Research before setteling on Mustache
+There are several templating scheme's we could have chosen. We had a quick discussion and most favor was for [mustache ](https://www.npmjs.com/package/mustache) as it  Mustashe is a well adtoped, lightweight library dedicated for templating.
+Alternatives considered:
+ * js template expression: This would have required all files to become js programs that needed to be imported and evaluated. Since a lot of files contain data format or code, we would have likely had quite a few ugly escaping cases.
+ * t4:  This would have reuquired visual studio to be installed on the users machine and therefore only work on windows. There were also some perf concerns and we would have to write and ship a standalone executable for the msbuild tasks so we can call it from javascript.
```

Comment:
* T4:  This would have required visual studio to be installed on the users machine and therefore only work on Windows. There were also some perf concerns and we would have to write and ship a standalone executable for the MSBuild tasks so we can call it from JavaScript.